framework/k4SimDelphes/src/k4SimDelphesAlg.h

```diff
@@ -0,0 +1,69 @@
+#ifndef _K4SIMDELPHESALG_H
+#define _K4SIMDELPHESALG_H
+
+#include "k4FWCore/DataHandle.h"
+#include "GaudiAlg/GaudiAlgorithm.h"
+#include "k4FWCore/DataWrapper.h"
+#include "k4FWCore/PodioDataSvc.h"
+
+#include "edm4hep/MCParticleCollection.h"
+#include "edm4hep/ReconstructedParticleCollection.h"
+
+ #include "k4SimDelphes/DelphesEDM4HepConverter.h"
+
+
+#include "modules/Delphes.h"
+
+namespace edm4hep {
+class MCParticleCollection;
+}
+
+/** @class k4SimDelphesAlg
+ *
+ *  Main Algorithm to run Delphes, getting MCParticle input, producing 
+ *  ReconstructedParticle output.
+**/
+class k4SimDelphesAlg : public GaudiAlgorithm {
+
+public:
+  k4SimDelphesAlg(const std::string& name, ISvcLocator* svcLoc);
+
+  virtual StatusCode initialize();
+  virtual StatusCode execute();
+  virtual StatusCode finalize();
+
+private:
+  /// Input from Generator
+  DataHandle<edm4hep::MCParticleCollection> m_InputMCParticles{"GenParticles", Gaudi::DataHandle::Reader, this};
+  /// Output from Delphes
+  DataHandle<edm4hep::ReconstructedParticleCollection> m_OutputRecParticles{"RecParticlesDelphes", Gaudi::DataHandle::Writer, this};
```

Comment:
How are the other output collections handled in this case? Are they configured via the tcl output configuration? How is the name given to the output reco particle collection from there handled in this case?